vllm/model_executor/models/aria.py

```diff
@@ -0,0 +1,734 @@
+        self.multi_modal_projector = build_mm_projector(config)
+        self.vocab_size = config.text_config.vocab_size
+        self.language_model = AriaMoELMModel(
+            vllm_config=vllm_config.with_hf_config(config.text_config),
+            prefix=maybe_prefix(prefix, "language_model.model"),
+        )
+        self.pad_token_id = (self.config.pad_token_id
+                             if self.config.pad_token_id is not None else -1)
+        self.unpadded_vocab_size = config.text_config.vocab_size
+        self.lm_head = ParallelLMHead(
+            self.unpadded_vocab_size,
+            config.text_config.hidden_size,
+            org_num_embeddings=self.language_model.org_vocab_size,
+            quant_config=quant_config,
+        )
+        logit_scale = getattr(config, "logit_scale", 1.0)
+        self.logits_processor = LogitsProcessor(self.unpadded_vocab_size,
+                                                self.vocab_size, logit_scale)
+        self.sampler = Sampler()
+
+    def forward(
+        self,
+        input_ids: torch.Tensor,
+        positions: torch.Tensor,
+        kv_caches: List[torch.Tensor],
+        attn_metadata: AttentionMetadata,
+        intermediate_tensors: Optional[IntermediateTensors] = None,
+        **kwargs: object,
+    ):
+        # 1. Extra the input embeddings
+        inputs_embeds = self.language_model.get_input_embeddings(input_ids)
+        pixel_values = kwargs.get("pixel_values", None)
+        pixel_mask = kwargs.get("pixel_mask", None)
+
+        # 2. Merge text and images
+        if pixel_values is not None:
+            if isinstance(pixel_values, torch.Tensor):
+                pixel_values = pixel_values.view(
+                    -1, *pixel_values.shape[-3:]).to(torch.bfloat16)
+                pixel_mask = pixel_mask.view(-1, *pixel_mask.shape[-2:])
+            elif isinstance(pixel_values, list):
+                if not all(x.shape[-3:] == pixel_values[0].shape[-3:]
+                           for x in pixel_values):
+                    raise ValueError("All images must be the same size")
+
+                pixel_values = [
+                    x.view(-1, *x.shape[-3:]).to(torch.bfloat16)
+                    for x in pixel_values
+                ]
+                pixel_values = torch.cat(pixel_values, dim=0)
+                pixel_mask = [x.view(-1, *x.shape[-2:]) for x in pixel_mask]
+                pixel_mask = torch.cat(pixel_mask, dim=0)
+            selected_image_feature, image_attn_mask = self.vision_tower(
+                pixel_values,
+                pixel_mask=pixel_mask,
+            )
+
+            image_features = self.multi_modal_projector(
+                selected_image_feature, attn_mask=image_attn_mask)
+
+            inputs_embeds = inputs_embeds.to(image_features.dtype)
+            inputs_embeds = merge_multimodal_embeddings(
+                input_ids, inputs_embeds, image_features,
+                self.config.image_token_index)
```

Comment:
Can you refactor the encoder processing with `process_mm_inputs`, `_parse_and_validate_image_input` and `get_input_embeddings` for code style alignment? You can refer to `llava`/`mllama`'s implementation:

https://github.com/vllm-project/vllm/blob/a111d0151ffed94582bec65635979e04e5b63676/vllm/model_executor/models/llava.py#L377-L378

https://github.com/vllm-project/vllm/blob/a111d0151ffed94582bec65635979e04e5b63676/vllm/model_executor/models/llava.py#L545-L553

In `_parse_and_validate_image_input`, we need to return a custom `TypedDict` wrapper named `AriaImagePixelInputs`.

Reply:
Yes!

Reply:
Done! Please take a look!